src/integrations/watchers/indexable-post-watcher.php

```diff
@@ -192,6 +192,15 @@ public function build_indexable( $post_id ) {
 
 			$post = $this->post->get_post( $post_id );
 
+			/*
+			 * Update whether an author has public posts.
+			 * For example this post could be set to Draft or Private,
+			 * which can influence if its author has any public posts at all.
+			 */
+			if ( $indexable ) {
+				$this->update_has_public_posts( $indexable );
+			}
+
```

Comment:
**Note for reviewer**: I moved this call out of the `updated_indexable` method to here to make sure that it is called whenever a post is set to a non-public post status.

Previously, this was only called when the post status was public (see the check on line 205). Which kind of defeats the purpose of  updating whether an author has public posts or not. This meant that setting a post to private or draft did not update the `has_public_posts` on the author indexable.